src/Shared/ProcessExtensions.cs

```diff
@@ -7,11 +7,11 @@
 using System.IO;
 using Microsoft.Build.Shared;
 
-namespace Microsoft.Build.Utilities
+namespace Microsoft.Build.Shared
 {
     internal static class ProcessExtensions
     {
-        public static void KillTree(this Process process, int timeout)
+        public static void KillTree(this Process process)
```

Comment:
Why did you eliminate the timeouts here? Seems good not to hang while trying to kill processes 😉

Reply:
I've added back the timeouts.

I guess the main reason for these timeouts is that descendent of the child proces may keep WaitForExit from returning. I'm making attempt at improving that in the BCL: https://github.com/dotnet/runtime/pull/48101 which would eliminate the need for these timeouts.